### ISSUE TITLE
Prepopulate run_id in schedule future_action_times

### ISSUE THREAD

**Issue by nnoto** [temporalio/temporal#4887]
**Is your feature request related to a problem? Please describe.**
I would like to track workflow executions via some unique identifier such as the run_id, including future workflow executions defined by a schedule. 

**Describe the solution you'd like**
Schedule information currently only provides a run_id for a workflow execution that has already begun. workflow_id isn't guaranteed to be unique when factoring in scheduled and on-demand workflows. Temporal documentation mentions a unique identifier of a workflow execution is the namespace + workflow_id + run_id.  Since future_action_times are provided for some portion of future schedule executions, it would be helpful to prepopulate the run_id as well is possible.

**Describe alternatives you've considered**
Including the schedule information (if applicable) that led to a workflow execution in the workflow description via list_workflows is helpful, but does not provide context for future workflow executions.

**Additional context**
Existing implementation: 
```
{  
    recent_actions {
        schedule_time {
            seconds: 1695312900
        }
        actual_time {
            seconds: 1695312900
            nanos: 405395930
        }
        start_workflow_result {
            workflow_id: "your-workflow-id-2023-09-21T16:15:00Z"
            run_id: "bae2a0b4-37b8-48f2-8a63-b6afc7d01b0f"
        }
    }
    future_action_times {
        seconds: 1695330900
    }
}
```
Proposed change:
```
{  
    recent_actions {
        schedule_time {
            seconds: 1695312900
        }
        actual_time {
            seconds: 1695312900
            nanos: 405395930
        }
        start_workflow_result {
            workflow_id: "your-workflow-id-2023-09-21T16:15:00Z"
            run_id: "bae2a0b4-37b8-48f2-8a63-b6afc7d01b0f"
        }
    }
    future_actions {
        workflow_id: <workflow_id>
        run_id: <workflow run_id>
        schedule_time {
            seconds: 1695330900
        }
    }
}
```

**Comment by dnr** [comment]
Currently `run_id` is generated at start time and I don't think we want to change that, so we can't predict the `run_id` ahead of time.

Can you use the workflow id (prefix + `-` + start time) or the value of the `TemporalScheduledStartTime` search attribute? The catch with using those is that those will use the time before jitter is added, while the time in `future_action_time` has the "actual" time including jitter. We could add a new field parallel to `future_action_time` that has the time without jitter so it'll match the workflow id

**Comment by nnoto** [comment]
@dnr Yes, using the workflow id is possible. The field parallel to future_action_time that has the time without jitter would be much appreciated if possible.

Has generating the run_ids based on the same method used for scheduled workflow ids (workflow prefix + - + start time) been considered? Currently the run_ids seem to be some UUID regardless of whether the workflow was run on-demand or scheduled.
The way I understand workflows is that the workflow id is essentially just the identifier for the action/routine comprising a set of activities and can be used for many executions of the workflow, so the workflow prefix should suffice for the workflow id even in the context of scheduled workflows. Since run ids are unique and represent an execution of a workflow, using the start time suffix seems more appropriate for the run_ids.

**Comment by dnr** [comment]
Run ids are explicitly represented as UUIDs in various database schemas so I'm afraid changing that isn't possible now. If it were a free-form string then yes, the timestamp would make sense. I think client-provided UUID run ids (where you could embed a timestamp, for example) could also violate certain assumptions in the rest of the system about uniqueness, so I don't think we can go there either.

The reason we put the timestamp in the workflow id itself instead of letting scheduled workflows reuse the workflow id is that schedules can be configured to allow scheduled workflows to run concurrently instead of enforcing exclusion. Only one workflow execution with the same workflow id can be running at once, and that's a basic invariant of Temporal.

That said, in configurations where we do enforce exclusion, we could reuse the workflow id (still with random run ids). That's requested in #4795.
